docs/plugin-protocol/tfplugin5.proto

```diff
@@ -831,6 +831,10 @@ message ListResource {
         // when include_resource_object is set to true, the provider should
         // include the full resource object for each result
         bool include_resource_object = 3;
+
+        // The maximum number of results that Terraform is expecting.
+        // The stream will stop, once this limit is reached.
+        int64 limit = 4;
```

Comment:
Checking my understanding:

For a provider implementation, is this an advisory/optional limit or a best-effort/required limit?

In other words: is the provider responsible for making a best-effort attempt to send no more than `limit` events? I want to say "yes, the provider is responsible for best-effort."